src/pages/VaultsPage/VaultsPage.tsx

```diff
@@ -13,7 +15,7 @@ const VaultsPage = () => {
   const closeModal = () => setModal('');
 
   const currentOrg = useCurrentOrg();
-  const vaults = useVaults(currentOrg.data?.id);
+  const { isLoading, data: vaults } = useVaults(Number(currentOrg.data?.id));
```

Comment:
This data refreshed correctly for me after creating the first vault, but not the second one

Reply:
I'm still review how best to do this. What happens is that the mutations is called directed on the backend when a new vault is created without using react-query. So I have to invalidate the cached query to trigger a re-render. Almost there.

Reply:
Works for me!

FWIW, seems like QueryClient.invalidateQueries is relatively [straightforward](https://react-query.tanstack.com/reference/QueryClient#queryclientinvalidatequeries) too, so we can look into it later on. I can imagine that in other cases it'd be good to just signal the need and let the refetching get triggered at the right time by the component that owns the query.